### DISCUSSION TITLE
I can't get the modal to open, postMessage failing

### DISCUSSION THREAD

**Discussion by Caio-Rezende** [WalletConnect/walletconnect-monorepo#2482] [Developer Support] [answered]
when initiating it is dispatching a postMessage that is being blocked thus causing the openModal to fail.

Failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('https://verify.walletconnect.com') does not match the recipient window's origin ('http://localhost:8080').

Already configured CORS in nginx with * and no good.

I would appreciate any insights, thanks.

**Comment by metabaseex** [comment]
I have the same problem,but no solution until now.
I do not use web3modal ,but use display_url message to show my own  modal window of qrcode.
the design of v2 is too low...with a lot of third party libs.

**Reply by glitch-txs** [reply]
Hi @metabaseex could open a different discussion for your case? So they can tracked separately :) also could you add the dependencies you're using + a code snippet there? Thanks 🙏

**Comment by glitch-txs** [comment]
Hi could you share which packages/version and framework/environment are you using?

**Reply by metabaseex** [reply]
API version:  V2
NodeJs 16.0+
Vue 3.2.13
not use web3modal ,but use 'display_uri' event to handle the customed modal show

**Reply by Caio-Rezende** [reply]
"@walletconnect/sign-client": "^2.7.7",
"@web3modal/standalone": "^2.4.1",
NodeJs v18.16.0

**Reply by glitch-txs** [reply]
Are you trying to initiate it on backend or vanilla js? Also could you show a snippet of your code?

There are a couple of examples in this repo as well:

https://github.com/WalletConnect/web3modal-examples

**Comment by Caio-Rezende** [comment, marked as answer]
The error was misleading, my problem was a missing await signal for the signClient.connect.

`
        const { uri, approval } = **_await_** signClient.connect({
`
